controller/backup_volume_controller.go

```diff
@@ -284,11 +284,27 @@ func (bvc *BackupVolumeController) reconcile(backupVolumeName string) (err error
 		return err
 	}
 
+	// Get `Time to live` after failed backup was marked as `Error` or `Unknown`,
+	// default: 180s and it is shorter than `Backupstore Poll Interval` by default, so failed backups should be deleted once the backup volume is updated.
+	failedBackupTTL, err := bvc.ds.GetSettingAsInt(types.SettingNameFailedBackupTTL)
+	if err != nil {
+		log.WithError(err).Errorf("failed to get %v setting from cluster, and it will skip the auto-deletion for the failed backups", types.SettingNameFailedBackupTTL)
+		// set failedBackupTTL to be 0 as disable failed backup auto-deletion
+		failedBackupTTL = 0
+	}
 	clustersSet := sets.NewString()
 	for _, b := range clusterBackups {
 		// Skip the Backup CR which is created from the local cluster and
 		// the snapshot backup hasn't be completed or pulled from the remote backup target yet
 		if b.Spec.SnapshotName != "" && b.Status.State != longhorn.BackupStateCompleted {
+			if b.Status.State == longhorn.BackupStateError || b.Status.State == longhorn.BackupStateUnknown {
```

Comment:
Is `longhorn.BackupStateUnknown` possibly a transient state? then cause an incorrect deletion.

Reply:
I think it is not a transient state.